### ISSUE TITLE
Voucher Types

### ISSUE THREAD

**Issue by drewroberts** [tipoff/vouchers#23]
We'll begin to use voucher types for restrictions on how vouchers can be applied during the checkout process.

**Comment by pdbreen** [comment]
Discounts are always pre-tax adjustments (tax is calculated on the discounted total).  Are vouchers always treated like cash (or a gift card) where the voucher amount is applied to the after tax total?  Or will there be a mix?  Some vouchers will be treated as cash and reduce the overall payment due, tax included - but others will be more like a discount that reduces the amount prior to tax calculation?  Iff the initial thinking is a voucher can be either - can we eliminate the mix by creating single use discounts instead leaving discounts always pre-tax and vouchers always post-tax?

**Comment by drewroberts** [comment]
> Can we eliminate the mix by creating single-use discounts instead, leaving Discounts always pre-tax and Vouchers always post-tax?

Yes, this would make things simpler. Vouchers will always be post-tax.

**Comment by drewroberts** [comment]
Should Voucher Types be switched to just an ENUM on the voucher model as we've been doing with types on other models?

The main things for types on the Voucher model was for various conditional application logic that may be needed like we do with Discounts.

Vouchers should also be able to be used for a number of participants or a specific product, etc. instead of a dollar amount. I had initially planned to use Types for that approach.

**Comment by drewroberts** [comment]
That's right. Thank you!
